junit_tests/TimeDeltaTest.java

```diff
@@ -0,0 +1,494 @@
+            HashMap<String, Object> kwargs = new HashMap<>();
+
+            kwargs.put("days", org.python.types.Int.getInt(1));
+            Assert.assertThrows(Exception.class, () ->  new TimeDelta(new Object[] { Int.getInt(1) }, kwargs));
+        }
+
+        { // Try too many arguments
+            HashMap<String, Object> kwargs = new HashMap<>();
+
+            Assert.assertThrows(Exception.class, () ->  new TimeDelta(new Object[] {
+                    Int.getInt(1),
+                    Int.getInt(2),
+                    Int.getInt(3),
+                    Int.getInt(4),
+                    Int.getInt(5),
+                    Int.getInt(6),
+                    Int.getInt(7),
+                    Int.getInt(8),
+                }, kwargs)
+            );
+        }
+    }
+
+    @Test
+    public void testNormalization() {
+        {
+            HashMap<String, Object> kwargs = new HashMap<>();
+
+            kwargs.put("microseconds", org.python.types.Int.getInt(1_000_000_000));
+            TimeDelta timeDelta = new TimeDelta(new Object[0], kwargs);
+
+            assertEquals(org.python.types.Int.getInt(1000), timeDelta.seconds);
+        }
+
+        {
+            HashMap<String, Object> kwargs = new HashMap<>();
+
+            kwargs.put("microseconds", org.python.types.Int.getInt(1_234_567_890));
+            TimeDelta timeDelta = new TimeDelta(new Object[0], kwargs);
+
+            assertEquals(org.python.types.Int.getInt(1234), timeDelta.seconds);
+            assertEquals(org.python.types.Int.getInt(567890), timeDelta.microseconds);
+        }
+        {
+            HashMap<String, Object> kwargs = new HashMap<>();
+
+            // as many seconds as a day plus one
+            kwargs.put("seconds", org.python.types.Int.getInt(86400 + 1));
+            TimeDelta timeDelta = new TimeDelta(new Object[0], kwargs);
+
+            assertEquals(org.python.types.Int.getInt(1), timeDelta.days);
+            assertEquals(org.python.types.Int.getInt(1), timeDelta.seconds);
+        }
+
+        {
+            HashMap<String, Object> kwargs = new HashMap<>();
+
+            // as many microseconds as 10 days
+            kwargs.put("microseconds", org.python.types.Int.getInt(86400L * 10_000_000L));
+            TimeDelta timeDelta = new TimeDelta(new Object[0], kwargs);
+
+            assertEquals(org.python.types.Int.getInt(10), timeDelta.days);
+        }
+
+        { // Negative seconds should decrease day
+            TimeDelta timeDelta = constructTimeDelta(0, -1, 0);
+
+            assertEquals(org.python.types.Int.getInt(-1), timeDelta.days);
+            assertEquals(org.python.types.Int.getInt(60*60*24-1), timeDelta.seconds);
+        }
+    }
+
+    @Test
+    public void testEquals() {
+        {
+            TimeDelta td0 = constructTimeDelta(1, 2, 3);
+            TimeDelta td1 = constructTimeDelta(1, 2, 3);
+
+            assertEquals(Bool.getBool(true), td0.__eq__(td1));
```

Comment:
This is a very minor thing and a question of preference and readability but you've used assertEquals instead of assertTrue in all of the comparison operation tests.

`assertEquals(Bool.getBool(true), td1.__eq__(td0));`

Could be changed into something like

`assertTrue((Boolean) td1.__eq__(td0).toJava());`

However, its not really a that much of an improvement in terms of readability so do whatever you prefer.